components/google-cloud/aiplatform/sdk/google_cloud_components/aiplatform/remote_runner.py

```diff
@@ -0,0 +1,154 @@
+# Copyright 2021 Google LLC. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Module for remote execution of AI Platform component."""
+
+import argparse
+import inspect
+import os
+from google.cloud import aiplatform
+from google.cloud import storage
+
+INIT_KEY = 'init'
+METHOD_KEY = 'method'
+
+
+# TODO() Add type-hinting the functions
+# TODO() Add explanation / exmaples and validation for kwargs
+def split_args(kwargs):
+  """Split args to match MB SDK input format for init, method and config."""
+  init_args = {}
+  method_args = {}
+  config_args = {}
+
+  for key, arg in kwargs.items():
+    if key.startswith(INIT_KEY):
+      init_args[key.split('.')[-1]] = arg
+    elif key.startswith(METHOD_KEY):
+      method_args[key.split('.')[-1]] = arg
+    else:
+      config_args[key.split('.')[-1]] = arg
+
+  return init_args, method_args, config_args
+
+# TODO() Add docstring and typ annotation. 
+def write_to_gcs(project, gcs_uri, text):
+  """Helper method to write resouce names to gcs for downstream components."""
+
+  gcs_uri = gcs_uri[len('gs://'):]
+  gcs_bucket, gcs_blob = gcs_uri.split('/', 1)
+
+  with open('resource_name.txt', 'w') as f:
+    f.write(text)
+
+  client = storage.Client(project=project)
+  bucket = client.get_bucket(gcs_bucket)
+  blob = bucket.blob(os.path.join(gcs_blob, 'resource_name.txt'))
+  blob.upload_from_filename('resource_name.txt')
+
+
+def read_from_gcs(project, gcs_uri):
```

Comment:
Should we read the resource name from GCS ourselves?
Perhaps we could use `{inputPath: input_name}` placeholder to get the local file with the data. Or maybe just use  `{inputValue: input_name}` to get the data itself.

Reply:
The gcs is used to store the rescue name, the plan is to switch this to using artifact / artifact uri. This code is initial prototype while we are setting the artifact input/output. made a tracker under b/184539511 to make sure we follow up.